### DISCUSSION TITLE
Can I load ReBarDxe with EFI Shell?

### DISCUSSION THREAD

**Discussion by rowanmoul** [xCuri0/ReBarUEFI#202] [Q&A]
I have a Dell PowerEdge R630 server that supports above 4G decoding already (at least that's what I interpret the ["Memory Mapped I/O above 4GB"](https://www.dell.com/support/manuals/en-ca/poweredge-r630/r630_om_pub/integrated-devices-details?guid=guid-0f237dca-ae8a-4972-985b-69da22be42ea&lang=en-us) option to be).
Unfortunately flashing a custom BIOS that isn't signed by Dell is a no-go without de-soldering the BIOS chip and using an external programmer, which is what someone did for their R730 that is in the list of working motherboards.

Previously on this server I have used an EFI Shell script to use the `load` command to load NVME DXE drivers that I compiled from the EDK2 project so that I could boot off an NVME drive. It worked by loading the driver, triggering a re-enumeration of block devices, and then executing the default bootx64.efi binary from the NVME drive.

I was hoping I could do a similar thing here, but unfortunately when I ran `load ReBarDxe.ffs` it says the file "is not an image".
Is there a way I can re-package this file so that it can be loaded like this? Or is there a some deeper reason this wouldn't work even if I did? My knowledge of UEFI is surface deep at best.

**Comment by xCuri0** [comment]
No you can't because it functions by hooking UEFI PCI allocation which happen early during POST
